baybe/utils/validation.py

```diff
@@ -66,3 +75,51 @@ def validator(self: Any, attribute: Attribute, value: Any) -> None:
 
 non_inf_float = _make_restricted_float_validator(allow_nan=True, allow_inf=False)
 """Validator for non-infinite floats."""
+
+
+def get_transform_objects(
+    objects: Sequence[_T],
+    df: pd.DataFrame,
```

Comment:
Related to my previous comment about the place of this utility, I would put `df` as the first argument. Argument: You can potentially search for no objects in a dataframe, but you cannot search for objects in an empty dataframe, making the objects attribute "more optional" than the dataframe.

Reply:
Sounds like a reasonable idea to me. I was anyway not entirely happy to have it in `validation` since it is not a pure validation function. With the change of having the dataframe first, the move makes sense. 

Note: I've also added a deprecation for the original method since it would otherwise be breaking change, strictly speaking (just a little name alias in the old namespace). 

Will leave this to @Scienfitz to close so that he can give his consent